### ISSUE TITLE
Jupyterbook-cache and notebook exclude_patterns not working as expected

### ISSUE THREAD

**Issue by scottyhq** [ICESAT-2HackWeek/website2022#71]
excluding a notebook from execution is not having any effect in CI, but works fine when running `jb build book` on the jupyterhub. 

https://github.com/ICESAT-2HackWeek/website2022/blob/adbad305e61af70153a64ba3883b575af90a3867/book/_config.yml#L37-L40

https://github.com/ICESAT-2HackWeek/website2022/runs/5562032944?check_suite_focus=true#step:3:1824)
```
reading sources... [100%] tutorials/jupyter                                    
[1824](https://github.com/ICESAT-2HackWeek/website2022/runs/5562032944?check_suite_focus=true#step:3:1824)
ERROR: Execution Failed with traceback saved in /home/runner/work/website2022/website2022/book/_build/html/reports/dataintegration-1.log
```

But it's really hard to figure out. I think we need some additional diagnostics in the github actions, for example to output `cat _config.yml` and disable caching by default for the book (it's mainly helpful later on when the number of notebooks grows)

**Comment by scottyhq** [comment]
I initially thought this had to do with caching https://github.com/ICESAT-2HackWeek/website2022/blob/adbad305e61af70153a64ba3883b575af90a3867/.github/actions/buildresources/action.yaml#L22, but then noticed the build log was different if running `jb build book` by itself or building via `./scripts/build_resources.sh`:

```
(notebook) jovyan@jupyter-scottyhq:~$ diff jb_build.log build_resources.log 
2,4c2,4
< Source Folder: /home/jovyan/website2022/book
< Config Path: /home/jovyan/website2022/book/_config.yml
< Output Path: /home/jovyan/website2022/book/_build/html
---
> Source Folder: /home/jovyan/website2022/scripts/../book
> Config Path: /home/jovyan/website2022/scripts/../book/_config.yml
> Output Path: /home/jovyan/website2022/scripts/../book/_build/html
13a14,15
> Executing: tutorials/DataIntegration/dataintegration-1 in: /home/jovyan/website2022/book/tutorials/DataIntegration
> Executing: tutorials/DataIntegration/dataintegration-2 in: /home/jovyan/website2022/book/tutorials/DataIntegration
15a18,19
> ERROR: Execution Failed with traceback saved in /home/jovyan/website2022/book/_build/html/reports/dataintegration-1.log
> ERROR: Execution Failed with traceback saved in /home/jovyan/website2022/book/_build/html/reports/dataintegration-2.log
```

So it seems the problem seems to be running `jb build RELATIVE_PATH`, which can cause things like exclude_patterns settings that use wildcards and paths to behave in unexpected ways. Probably the best approach is to always run `jb build book/`

**Comment by scottyhq** [comment]
closed by #73
